src/components/LanguageSelector/LanguageSelector.vue

```diff
@@ -0,0 +1,50 @@
+<template>
+  <div>
+    <template v-for="{ locale, name } in languages">
+      <NuxtLink
+        v-if="locale !== $i18n.locale"
+        :key="locale"
+        :to="getLocalePath(locale)"
+        :hreflang="locale"
+        class="app-menu__link"
+        @click="$emit('close')"
+      >
+        <SvgIcon name="world-icon" class="app-menu__icon" />
+        <span class="app-menu__link-name">
+          {{ name }}
+        </span>
+      </NuxtLink>
+    </template>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { storeToRefs } from "pinia";
+import { useStore } from "~/stores/store";
+import { useHistoryStore } from "~/stores/history";
+
+const languages = [
+  { locale: "en", name: "english" },
+  { locale: "nl", name: "nederlands" },
+];
+
+const history = useHistoryStore();
+const store = useStore();
+const { selection } = storeToRefs(store);
+const { genericRoute, buildingRoute, spaceRoute } = useLocaleRoute();
+function getLocalePath(locale: string) {
+  const localisedRouteName = history.currentPageRoute.name;
+  const genericRouteName = localisedRouteName.split("___")[0];
```

Comment:
Maybe this was already in, but this is hard to read.
Can you destructure `genericRouteName.split("-")` and give the array parts meaningful names instead of `parts[1]` etc, so it's easier to understand?
I also don't understand why `parts.length < 3` is a condition. (maybe I wrote this 4 years ago :) don't know, but maybe now's a good time to improveet this)

Reply:
I guess it's more readable now, please take a look